pandas/io/json/_json.py

```diff
@@ -352,7 +353,10 @@ def __init__(
             )
             raise ValueError(msg)
 
+        obj = obj.copy()
```

Comment:
This copy call should only be called when needed

Reply:
It was in the code a couple of lines lower, I just moved it up. I assume putting a copy call into the new `if` statement would be overkill as it will be followed by the old one.

Reply:
Could you merge in main? I think the copy was moved since this PR was opened

Reply:
Rebased on main, deleted the copy call completely as it is done anyway inside the `set_default_names`